lib/rules/one-var.js

```diff
@@ -272,6 +276,87 @@ module.exports = {
             return true;
         }
 
+        /**
+         * Fixer to join VariableDeclaration's into a single declaration
+         * @param   {VariableDeclaration} declarations The `VariableDeclaration` to join
+         * @returns {Function}                         The fixer function
+         */
+        function joinDeclarations(declarations) {
+            const declaration = declarations[0];
+
+            return fixer => {
+                const type = sourceCode.getTokenBefore(declaration);
+                const prevSemi = sourceCode.getTokenBefore(type);
```

Comment:
I think this will work incorrectly when there is another statement between the two `VariableDeclaration` nodes. For example:

```js
/* eslint one-var: [error, { initialized: "always" }] */

var foo = 1; somethingElse(); var bar = 2;
```

This currently gets fixed to `var foo = 1; somethingElse(),  bar = 2;`, which is not correct.

To avoid this issue, I think we should add a check before autofixing to make sure the two `VariableDeclaration` statements are consecutive.